x-pack/plugins/security/server/routes/session_management/delete_all.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import typeDetect from 'type-detect';
+import { schema } from '@kbn/config-schema';
+import { RouteDefinitionParams } from '..';
+
+/**
+ * Defines routes required for the deleting of all sessions.
+ */
+export function defineDeleteAllSessionsRoutes({ router, getSession }: RouteDefinitionParams) {
+  router.delete(
+    {
+      path: '/internal/security/session/_all',
+      validate: {
+        query: schema.maybe(
+          schema.object(
+            {
+              // `providerType` is actually required when any other options are specified, but such schema isn't
+              // currently supported by the Core for `query` string parameters. To workaround that we mark this property
+              // as optional and do custom validation instead: https://github.com/elastic/kibana/issues/92201
+              providerType: schema.maybe(schema.string()),
+              providerName: schema.maybe(schema.string()),
+              username: schema.maybe(schema.string()),
+            },
+            {
+              validate(value) {
+                if (typeof value?.providerType !== 'string' && Object.keys(value).length > 0) {
+                  return `[request query.providerType]: expected value of type [string] but got [${typeDetect(
+                    value?.providerType
+                  )}]`;
+                }
+              },
+            }
+          )
+        ),
+      },
+      options: { tags: ['access:sessionManagement'] },
```

Comment:
__question:__ do we have any precedent\ability to relax superuser-only requirement and, for example, give access to `kibana_admin`?

Reply:
> Are there benefits to relaxing this requirement that I'm overlooking?

Nothing concrete yet, I'm just not very happy that we require a "cluster superpower" to invalidate Kibana-specific sessions and `Kibana Admin` sounded like a role that could be assumed to be responsible for all Kibana-specific things. I'm totally fine to keep it as is right now, but if we can eventually have a Kibana Operator role or something, I'd vote for it instead.

Reply:
> > Are there benefits to relaxing this requirement that I'm overlooking?
> 
> Nothing concrete yet, I'm just not very happy that we require a "cluster superpower" to invalidate Kibana-specific sessions and `Kibana Admin` sounded like a role that could be assumed to be responsible for all Kibana-specific things. I'm totally fine to keep it as is right now, but if we can eventually have a Kibana Operator role or something, I'd vote for it instead.

Yeah what you're saying makes sense. My fear is that `kibana_admin` would be granted to too many folks, especially without an "operator" role readily available. More than happy to revisit this in the future!

Reply:
> My fear is that kibana_admin would be granted to too many folks, especially without an "operator" role readily available. 

Yeah, that's a good point, let's see how it goes with superuser then.